src/odemis/util/angleres.py

```diff
@@ -512,13 +512,14 @@ def Rectangular2Polar(data, output_size, colormap=None, hole=False):
 
     # Calculate the mirror mask in rectangular phi/theta representation
     # Note: Everything outside of mask is 0
-    mask = _CreateMirrorMaskRectangular(data, hole=hole)
+    mask = _CreateMirrorMaskRectangular(data, hole=True)
 
     # Get the upper/lower intensity limit, by checking for a fixed amount of outliers.
-    data[numpy.isnan(data)] = 0  # TODO small hack to handle potential nans in data (present in normalized img)
-    lim1, lim2 = img.getOutliers(data[mask], outliers=1/256)  # 1/256 seems to work well for most datasets
-    data_masked = data.copy()
+    # In case of NaNs in projection, remove NaNs for calculating histogram limits, but keep for plotting.
+    mask_outliers = mask & ~numpy.isnan(data)  # create new mask, which masks NaNs in data
+    lim1, lim2 = img.getOutliers(data[mask_outliers], outliers=1/256)  # 1/256 seems to work well for most datasets
 
+    data_masked = data.copy()
```

Comment:
Actually, that copy is probably not needed anymore, as data_masked is not modified. => just write `data_masked = ma.masked_array(data, ~mask)`